### ISSUE TITLE
SNI support for TLS handshake

### ISSUE THREAD

**Issue by hellais** [ooni/probe-engine#105]
@darkk commented on [Wed Aug 17 2016](https://github.com/ooni/probe/pull/588)

Sometimes https traffic is blocked on SNI basis, so I've written some code to test that.

It should be probably merged after 2.0.0 release or rebased onto 2.0.

---

@hellais commented on [Tue Aug 30 2016](https://github.com/ooni/probe/pull/588#issuecomment-243416158)

Minor comments to this test. I will put the merge for this on hold until we get the 2.0.0 branch merged into master.

Good work 👍 

---

@hellais commented on [Tue Sep 19 2017](https://github.com/ooni/probe/pull/588#issuecomment-330482203)

I think it would be cool to land this into master. It also has a useful fix to `NetTest` class that at minimum should make it's way into master.

Given the fact tls_handshake is experimental anyways, I don't see any harm in making experimental changes in it.

It needs to be rebased though.

**Comment by bassosimone** [comment]
@hellais as far as I understand, I implemented this in netx already.

**Comment by bassosimone** [comment]
Yeah, upon re-reading the diff, this has been implemented in netx already.
